File: src/altair_py_tests/altair_py_tests/modules/rclnode_test_node.py
import rclpy
from rclpy.node import Node
import std_msgs.msg as std_msgs 



class RclNodeTestNode(Node):

    def __init__(self) -> None:
        super().__init__('RclNodeTestNode')

        self.declare_parameter('pub_msg', rclpy.Parameter.Type.STRING)
        self.declare_parameter('pub_topic', rclpy.Parameter.Type.STRING)
        self.declare_parameter('sub_topic', rclpy.Parameter.Type.STRING)

        self.pub_msg    = self.get_parameter('pub_msg').value
        self.pub_topic  = self.get_parameter('pub_topic').value
        self.sub_topic  = self.get_parameter('sub_topic').value

        self.subscriber = self.create_subscription(
            msg_type    = std_msgs.String,
            topic       = self.sub_topic,
            callback    = self.subCallback,
            qos_profile = 1000
        )

        self.publisher = self.create_publisher(
            msg_type    = std_msgs.String,
            topic       = self.pub_topic,
            qos_profile = 1000
        )

        self.pub_timer = self.create_timer(1., self.pubTimerCallback)



    def subCallback(self, msg:std_msgs.String) -> None:
        self.get_logger().info(f'[{self.sub_topic}]: {msg.data}')



    def pubTimerCallback(self) -> None:
        msg         = std_msgs.String()
        msg.data    = self.pub_msg
        self.publisher.publish(msg)
        
        self.get_logger().info(f'message sent to {self.pub_topic}')

File: src/altair_cpp_tests/src/altair_cpp_tests/rclnode_test_node.cpp
#include "altair_cpp_tests/rclnode_test_node.hpp"



altair::RclNodeTestNode::RclNodeTestNode() : rclcpp::Node("RclNodeTestNode") {
    this->declare_parameter("pub_msg", rclcpp::PARAMETER_STRING);
    this->declare_parameter("pub_topic", rclcpp::PARAMETER_STRING);
    this->declare_parameter("sub_topic", rclcpp::PARAMETER_STRING);

    this->pub_msg   = this->get_parameter("pub_msg").as_string();
    this->pub_topic = this->get_parameter("pub_topic").as_string();
    this->sub_topic = this->get_parameter("sub_topic").as_string();

    this->subscriber = this->create_subscription<std_msgs::msg::String>(
        this->sub_topic.c_str(),
        1000,
        std::bind(&altair::RclNodeTestNode::subCallback, this, _1)
    );

    this->publisher = this->create_publisher<std_msgs::msg::String>(
        this->pub_topic.c_str(),
        1000
    );

    this->pub_timer = this->create_wall_timer(
        1s,
        std::bind(&altair::RclNodeTestNode::pubTimerCallback, this)
    );
}



altair::RclNodeTestNode::~RclNodeTestNode() {
    RCLCPP_INFO(this->get_logger(), "RclNodeTestNode destroyed");
}



void altair::RclNodeTestNode::subCallback(const std_msgs::msg::String &msg) {
    RCLCPP_INFO(this->get_logger(), "[%s]: %s", this->sub_topic.c_str(), msg.data.c_str());
}



void altair::RclNodeTestNode::pubTimerCallback() {
    auto msg = std_msgs::msg::String();
    msg.data = this->pub_msg;
    this->publisher->publish(msg);
    RCLCPP_INFO(this->get_logger(), "message sent to %s", this->pub_topic.c_str());
}